### PR TITLE
(GH-123) First use HTTP HEAD and GET as a fallback

### LINKED ISSUE #123: Use HEAD request, rather than GET request
In this location:

https://github.com/gep13/chocolatey-vscode/blob/develop/src/Chocolatey.Language.Server/UriExtensions.cs#L17

This will mean that no response is required to be sent to the client, and should speed things up.

Should be possible using HttpWebRequest, which can set the HEAD method type:

https://docs.microsoft.com/en-us/dotnet/api/system.net.httpwebrequest.method?view=netframework-4.7.2

### PR DESCRIPTION
**Pull request by mkevenaar**
Fixes #123 

This switches to HTTP HEAD requests, if that failed, it uses GET requests.

### CODE REVIEW

File: src/Chocolatey.Language.Server/UriExtensions.cs [hunk 1]
```diff
@@ -12,13 +12,15 @@ public static class UriExtensions
         ///   Tries to validate an URL
         /// </summary>
         /// <param name="url">Uri object</param>
-        public static bool IsValid(this Uri url)
+        /// <param name="get">Bool Switch to GET request.</param>
+        public static bool IsValid(this Uri url, bool get = false)
```
Comment (AdmiringWorm) [inline src/Chocolatey.Language.Server/UriExtensions.cs]: Maybe a better name for the parameter would be `useGetRequest`, or `useGetMethod`?

Just a suggestion though.

Comment (mkevenaar) [reply src/Chocolatey.Language.Server/UriExtensions.cs]: Good catch! let me fix that real quick!

Comment (mkevenaar) [reply src/Chocolatey.Language.Server/UriExtensions.cs]: Done

**Comment by gep13** [comment]
@mkevenaar I have rebased and merged this here:

https://github.com/gep13/chocolatey-vscode/commit/7b90b472c5caaaf6d8bea98db8f554ca6808c44b

**Comment by gep13** [comment]
@mkevenaar your changes have been merged, thanks for your contribution 👍

**Pull request closed without merging**